fuel-core/src/executor.rs

```diff
@@ -592,11 +647,13 @@ impl Executor {
     /// In validation mode, verify the proposed utxo ids on contract inputs match the expected values.
     fn compute_contract_input_utxo_ids(
         &self,
-        tx: &mut Transaction,
-        mode: &ExecutionMode,
+        tx: &mut ExecutionType<&mut Transaction>,
         db: &Database,
     ) -> Result<(), Error> {
-        if let Transaction::Script { inputs, .. } = tx {
+        // Get the kind of execution.
+        let execution_kind = tx.to_kind();
+
+        if let Transaction::Script { inputs, .. } = tx.deref_mut() {
```

Comment:
this ExecutionType wrapper could also be removed since the tx is always mutable and only the kind is really needed.

Reply:
I went with a different approach which I think actually captures the fact that the inputs are only mutable in production.